R/manual_layer_delayed.R

```diff
@@ -58,6 +59,53 @@ delayedSQL <- function(query, name, namespace) {
   }, local=TRUE)
 }
 
+##' Define a generic UDF to be executed within a task graph
+##'
+##' @param namespace The TileDB-Cloud namespace to charge the query to
+##'
+##' @param udf An R function. Arguments are specified separately via \code{args}.
+##' One of \code{udf} and \code{registered_udf_name} must be non-null.
+##'
+##' @param registered_udf_name Name of a registered UDF, of the form \code{namespace/udfname}.
+##' Arguments are specified separately via \code{args}.  One of \code{udf} and
+##' \code{registered_udf_name} must be non-null.
+##'
+##' @param args Can be provided here with \code{c <- delayed(function(...) { sum(...) }, args=list(a,b))}
+##' or ##' separately with \code{c <- delayedUDF(...)` and later \code{c$args <- list(a,b)}.
+##'
+##' @param name Optional -- e.g. \code{a} or \code{b}. If omitted, it defaults to a UUID.
+##'
+##' @return The return value from the UDF as an R object.
+##'
+##' @family {manual-layer functions}
+##' @export
+delayedUDF <- function(namespace, udf=NULL, registered_udf_name=NULL, args=NULL, name=NULL)
```

Comment:
In Python, by default the namespace is the currently-logged-in user’s charging namespace. Is this something you can do with the R API? If not, might it still make sense to put `namespace` later in the args list, so that the call looks more like:

`myUDF <- delayedUDF(myFn, args=...)`

Reply:
@thetorpedodog indeed :)

Two things to be done on follow-on PRs:

* Take the default namespace form the currently-logged-in user's charging namespace.
* Audit where in the args list `namespace` appears and try to uniformize it.

I want to do these before the 0.0.7 release, so there won't be backward-incompability issues.